pkgs/development/libraries/duckdb/default.nix

```
@@ -1,26 +1,65 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
 , cmake
+, ninja
+, openssl
+, openjdk11
+, unixODBC
+, withFts ? true
+, withHttpFs ? true
+, withIcu ? true
+, withJdbc ? false
+, withOdbc ? false
+, withParquet ? true
+, withRest ? true
+, withTpcDs ? true
+, withTpcH ? true
+, withVisualizer ? true
 }:
 
+let
+  enableFeature = yes: if yes then "ON" else "OFF";
+in
 stdenv.mkDerivation rec {
   pname = "duckdb";
-  version = "0.3.1";
+  version = "0.3.2";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-apTU7WgKw/YEnT4maibyffqOrCoVwHPOkNINlAmtYYI=";
+    sha256 = "sha256-F5YOqDeY3rgcnuu5SNqOfUxhsaXgqvdJZTnD1unI0tc=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  patches = [ ./version.patch ];
+  postPatch = ''
+    substituteInPlace CMakeLists.txt --subst-var-by DUCKDB_VERSION "v${version}"
+  '';
+
+  cmakeFlags = [
+    "-DBUILD_FTS_EXTENSION=${enableFeature withFts}"
+    "-DBUILD_HTTPFS_EXTENSION=${enableFeature withHttpFs}"
+    "-DBUILD_ICU_EXTENSION=${enableFeature withIcu}"
+    "-DBUILD_ODBC_DRIVER=${enableFeature withOdbc}"
+    "-DBUILD_PARQUET_EXTENSION=${enableFeature withParquet}"
+    "-DBUILD_REST_EXTENSION=${enableFeature withRest}"
+    "-DBUILD_TPCDS_EXTENSION=${enableFeature withTpcDs}"
+    "-DBUILD_TPCH_EXTENSION=${enableFeature withTpcH}"
+    "-DBUILD_VISUALIZER_EXTENSION=${enableFeature withVisualizer}"
+    "-DJDBC_DRIVER=${enableFeature withJdbc}"
```

As usual, I suggest not introducing excessive options
```suggestion
    "-DBUILD_FTS_EXTENSION=ON"
    "-DBUILD_HTTPFS_EXTENSION=${enableFeature withHttpFs}"
    "-DBUILD_ICU_EXTENSION=ON"
    "-DBUILD_ODBC_DRIVER=${enableFeature withOdbc}"
    "-DBUILD_PARQUET_EXTENSION=ON"
    "-DBUILD_REST_EXTENSION=ON"
    "-DBUILD_TPCDS_EXTENSION=ON"
    "-DBUILD_TPCH_EXTENSION=ON"
    "-DBUILD_VISUALIZER_EXTENSION=ON"
    "-DJDBC_DRIVER=${enableFeature withJdbc}"
```

---

Yup, thanks.